symbolic-base/src/ZkFold/Symbolic/Data/Int.hs

```diff
@@ -0,0 +1,216 @@
+{-# LANGUAGE AllowAmbiguousTypes  #-}
+{-# LANGUAGE TypeOperators        #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+
+module ZkFold.Symbolic.Data.Int where
+
+import           Control.DeepSeq
+import qualified Data.Bool                        as Haskell
+import           Data.Constraint                  (withDict)
+import           Data.Constraint.Nat
+import           Data.Kind                        (Type)
+import           GHC.Generics                     (Generic)
+import           Prelude                          (Integer, type (~), ($), (.))
+import qualified Prelude                          as Haskell hiding ((-))
+import           Test.QuickCheck                  (Arbitrary (..))
+
+import           ZkFold.Base.Algebra.Basic.Class  hiding (Euclidean (..))
+import           ZkFold.Base.Algebra.Basic.Number
+import           ZkFold.Base.Data.Vector          (Vector (..))
+import           ZkFold.Symbolic.Class
+import           ZkFold.Symbolic.Data.Bool
+import           ZkFold.Symbolic.Data.Class       (SymbolicData)
+import           ZkFold.Symbolic.Data.Combinators
+import           ZkFold.Symbolic.Data.Conditional
+import           ZkFold.Symbolic.Data.Eq
+import           ZkFold.Symbolic.Data.Ord
+import           ZkFold.Symbolic.Data.UInt
+import           ZkFold.Symbolic.Interpreter      (Interpreter (..))
+
+
+newtype Int (n :: Natural) (r :: RegisterSize) (c :: (Type -> Type) -> Type) = Int { uint :: UInt (n+1) r c}
```

Comment:
I think it's better to use `UInt n r c` to reuse same type-level constraints from `UInt` and to make the API more intuitive. Compare it with usual `uint32_t`/`int32_t` from C, for example.